crates/dbsp/src/circuit/operator_traits.rs

```diff
@@ -26,6 +28,9 @@ pub trait Operator: 'static {
         None
     }
 
+    /// Reports metrics about the current operator
+    fn metrics(&self, _global_id: &GlobalNodeId) {}
```

Comment:
A more future-proof design might be passing a handle with methods like `metric_gauge()`, `metric_histogram()`.